mmcv/runner/hooks/lr_updater.py

```diff
@@ -397,10 +399,14 @@ def __init__(self,
             '"target_ratio" must be list or tuple of two floats'
         assert 0 <= step_ratio_up < 1.0, \
             '"step_ratio_up" must be in range [0,1)'
+        assert 0 < gamma <= 1, \
+            '"gamma" must be in range (0, 1]'
 
         self.target_ratio = target_ratio
         self.cyclic_times = cyclic_times
         self.step_ratio_up = step_ratio_up
+        self.gamma = gamma
+        self.max_iter_per_phase = None
```

Comment:
It is duplicated with line 428.

Reply:
It's duplicated on purpose, since I followed the convention to always declare all fields in the `__init__` method. If this convention is not generally followed in mmcv, I can remove it.

Reply:
Agree with you